src/formatDate.js

```diff
@@ -8,7 +8,53 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let fromSeparator = '';
+
+  for (const arg of fromFormat) {
+    if (arg !== 'YYYY' || arg !== 'YY' || arg !== 'MM' || arg !== 'DD') {
+      fromSeparator = arg;
+    }
```

Comment:
The condition `arg !== 'YYYY' || arg !== 'YY' || arg !== 'MM' || arg !== 'DD'` is incorrect because it will always evaluate to `true`. This is because `arg` cannot be equal to all these values at the same time. You should use `&&` instead of `||` to check if `arg` is not equal to any of these values.